File: fibonacci-factorial/src/recurrent-functions.js
const axios = require("axios");

function factorialRecursive(n) {
  if (n === 0) {
    return 1;
  }
  return n * factorialRecursive(n - 1);
}

function fibonacciRecursive(n) {
  if (n <= 1) {
    return n;
  }
  return fibonacciRecursive(n - 1) + fibonacciRecursive(n - 2);
}

const functionType = process.argv[2];
const number = parseInt(process.argv[3]);

if (!["factorial", "fibonacci"].includes(functionType) || isNaN(number)) {
  console.error(
    "Niepoprawne parametry. Użyj: node recurrent-functions.js [factorial|fibonacci] [liczba]"
  );
  process.exit(1);
}

let result;
if (functionType === "factorial") {
  result = factorialRecursive(number);
} else if (functionType === "fibonacci") {
  result = fibonacciRecursive(number);
}

const sendData = async () => {
  try {
    const response = await axios.post(`http://localhost:3333/${functionType}`, {
      result: result,
    });
    console.log(response.data);
  } catch (error) {
    console.error("Wystąpił błąd podczas wysyłania żądania:", error);
  }
};

sendData();

File: fibonacci-factorial/src/recurrent-functions.py
import requests
import sys

def factorial_recursive(n):
    if n == 0:
        return 1
    return n * factorial_recursive(n - 1)

def fibonacci_recursive(n):
    if n <= 1:
        return n
    return fibonacci_recursive(n - 1) + fibonacci_recursive(n - 2)

function_type = sys.argv[1]
number = int(sys.argv[2])

if function_type not in ["factorial", "fibonacci"] or not number:
    print("Niepoprawne parametry. Użyj: python recurrent-functions.py [factorial|fibonacci] [liczba]")
    sys.exit(1)

result = None
if function_type == "factorial":
    result = factorial_recursive(number)
elif function_type == "fibonacci":
    result = fibonacci_recursive(number)

def send_data():
    try:
        response = requests.post(f"http://localhost:3333/{function_type}", json={"result": result})
        print(response.json())
    except requests.exceptions.RequestException as error:
        print("Wystąpił błąd podczas wysyłania żądania:", error)

send_data()